### DISCUSSION TITLE
Show 2FA information on package versions

### DISCUSSION THREAD

**Discussion by Eomm** [npm/feedback#706] [Website] [answered]
Hi,
when I select a package to add to my project, I would like to know if the module I choose is released by using 2FA or not.
This would help me to decide if add it to my project or select another one.

I think it would be useful to see this information on the package's version page.

It would be terrific to see if the release has been done by using a TOKEN with or without 2FA as well.

What do you think?

**Comment by ljharb** [comment]
Wouldn’t this potentially leak which packages do not have 2FA required, and also which users do not have 2FA enabled?

**Reply by Eomm** [reply]
Yes, but if orgs/maintainer can opt-in this feature.

So the status would be:

- hidden: the org/maintainer does not show this information
- visible: show the "released with 2FA" badge

That being said, the first top 100 maintainers have already forced 2FA, but it is not clear which are the top 100 maintainers looking at the user profile only

**Reply by ljharb** [reply]
I suppose if it was opt-in it’d be fine, but then wouldn’t it likely not show up on enough packages to be a useful signal?

**Reply by Eomm** [reply]
Yes, if nobody uses this feature the status would remain as it is.

Is a pool something that would help to understand if this feature is worth the effort?

**Comment by MylesBorins** [comment, marked as answer]
This is somewhat of a duplication of https://github.com/npm/feedback/discussions/588

We don't discourage folks from publishing via an automation tokens, and in many ways publishing from CI can be more secure than publishing from a local machine with 2FA.

Showing 2FA adoption on packages is not something that we are interested in doing as it leaks private information that could be used to target specific packages and customers
